reasoner/resolution/resolver/ConjunctionResolver.java

```diff
@@ -0,0 +1,308 @@
+            respondToUpstream(new Response.Fail(fromUpstream), iteration);
+            return;
+        }
+
+        responseProducer.removeDownstreamProducer(fromDownstream.sourceRequest());
+        tryAnswer(fromUpstream, responseProducer, iteration);
+    }
+
+    @Override
+    protected void initialiseDownstreamActors() {
+        LOG.debug("{}: initialising downstream actors", name());
+        Set<Concludable> concludables = Iterators.iterate(Concludable.create(conjunction))
+                .filter(c -> c.getApplicableRules(conceptMgr, logicMgr).hasNext()).toSet();
+        Set<Retrievable> retrievables = Retrievable.extractFrom(conjunction, concludables);
+        Set<Resolvable<?>> resolvables = new HashSet<>();
+        resolvables.addAll(concludables);
+        resolvables.addAll(retrievables);
+
+        plan.addAll(planner.plan(resolvables));
+        iterate(plan).forEachRemaining(resolvable -> {
+            downstreamResolvers.put(resolvable, registry.registerResolvable(resolvable));
+        });
+
+        // TODO just adding negations at the end, but we will want to include them in the planner
+        for (Negation negation : conjunction.negations()) {
+            Negated negated = new Negated(negation);
+            plan.add(negated);
+            downstreamResolvers.put(negated, registry.negated(conjunction, negated));
+        }
+    }
+
+    private ResponseProducer mayUpdateAndGetResponseProducer(Request fromUpstream, int iteration) {
+        if (!responseProducers.containsKey(fromUpstream)) {
+            responseProducers.put(fromUpstream, responseProducerCreate(fromUpstream, iteration));
+        } else {
+            ResponseProducer responseProducer = responseProducers.get(fromUpstream);
+            assert responseProducer.iteration() == iteration ||
+                    responseProducer.iteration() + 1 == iteration;
+
+            if (responseProducer.iteration() + 1 == iteration) {
+                // when the same request for the next iteration the first time, re-initialise required state
+                ResponseProducer responseProducerNextIter = responseProducerReiterate(fromUpstream, responseProducer, iteration);
+                responseProducers.put(fromUpstream, responseProducerNextIter);
+            }
+        }
+        return responseProducers.get(fromUpstream);
+    }
+
+    @Override
+    protected ResponseProducer responseProducerCreate(Request fromUpstream, int iteration) {
+        LOG.debug("{}: Creating a new ResponseProducer for request: {}", name(), fromUpstream);
+
+//        ResourceIterator<AnswerState.UpstreamVars.Derived> upstreamAnswers = toUpstreamAnswers(
+//                fromUpstream, compatibleBoundAnswers(conceptMgr, conjunction, fromUpstream.partialAnswer().conceptMap()));
+//
+        ResponseProducer responseProducer = new ResponseProducer(Iterators.empty(), iteration);
+        assert !plan.isEmpty();
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Use of identifier that is a keyword in later versions of Java (NM_FUTURE_KEYWORD_USED_AS_IDENTIFIER)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-use-of-identifier-that-is-a-keyword-in-later-versions-of-java-nm-future-keyword-used-as-identifier).
The identifier is a word that is reserved as a keyword in later versions of Java, and your code will need to be changed in order to compile it in later versions of Java.